server/lomas_server/dp_queries/dp_libraries/opendp.py

```diff
@@ -79,37 +81,34 @@ def get_lf_domain(metadata, by_config):
         )
         series_domains.append(series_domain)
 
-    lf_domain = dp.domains.lazyframe_domain(series_domains)
-
     # Margins
     # TODO Check lengths vs. keys for public info -> not in doc anymore.
 
     # Global params
-    params = {}
-    params["max_num_partitions"] = metadata["max_ids"]
-    if "rows" in metadata:
-        params["max_partition_length"] = metadata["rows"]
-    if len(params) > 0:
-        lf_domain = dp.domains.with_margin(
-            lf_domain,
-            by=[],
-            public_info="lengths",
-            **params,
-        )
-
-    # Grouping logic (margin adaptation)
+    params = _get_global_params(metadata)
     if by_config:
         params = _update_params_by_grouping(metadata, by_config, params)
-        lf_domain = dp.domains.with_margin(
-            lf_domain,
-            by=by_config,
-            public_info="lengths",
-            **params,
-        )
+    else:
+        by_config = []
+
+    lf_domain = dp.domains.with_margin(
+        dp.domains.lazyframe_domain(series_domains),
+        by=by_config,
+        public_info="lengths",
+        **params,
+    )
 
     return lf_domain
 
 
+def _get_global_params(metadata):
+    params = {}
+    params["max_num_partitions"] = 1
+    params["max_partition_length"] = metadata["rows"]
+
+    return params
+
+
 def _update_params_by_grouping(metadata, by_config, params):
```

Comment:
Not sure we need the underscore for these functions? get_lf_domain is also only used "internally" in this file for example. Only minor detail / matter of taste

Reply:
make sense